crates/red_knot_python_semantic/src/module_resolver/resolver.rs

```diff
@@ -202,13 +201,10 @@ impl SearchPaths {
 
             let search_path = SearchPath::custom_stdlib(db, &custom_typeshed)?;
 
-            (ResolvedTypeshedVersions::Custom(parsed), search_path)
+            (Some(parsed), search_path)
         } else {
             tracing::debug!("Using vendored stdlib");
-            (
-                ResolvedTypeshedVersions::Vendored(vendored_typeshed_versions()),
```

Comment:
I think it's fine to defer validating the vendored typeshed versions. They should be valid and whether we panic here or when trying to resolve the first module doesn't make a real difference. It results in a panic.

Reply:
Okay, there's a 1% performance drop. That doesn't seem worth it. 

I'm just gonna make `vendored_typeshed_versions` always parse the file without caching. This is an overall simplification and shouldn't hurt performance except in the case when we reconstruct the `SearchPathSettings` because the project configuration has changed.... but that's a very slow path anyway